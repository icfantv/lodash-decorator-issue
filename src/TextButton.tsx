import * as React from 'react'
import { BindAll } from 'lodash-decorators'

export interface ITextButtonProps {
  onClick: () => void
}

@BindAll()
export class TextButton extends React.Component<ITextButtonProps> {
  public handleButtonClick(): void {
    this.props.onClick()
  }

  public render(): React.ReactElement<TextButton> {
    return <button onClick={this.handleButtonClick}>Click Me!</button>
  }
}
