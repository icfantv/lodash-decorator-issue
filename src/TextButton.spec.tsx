import * as React from 'react'
import { create } from 'react-test-renderer'

import { TextButton } from './TextButton'

describe('the <TextButton/> component', () => {
  test('it should render the button', () => {
    const tree = create(<TextButton onClick={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
