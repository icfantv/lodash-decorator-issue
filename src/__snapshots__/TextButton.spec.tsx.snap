// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <TextButton/> component it should render the button 1`] = `
<button
  onClick={[Function]}
>
  Click Me!
</button>
`;
